test/e2e/specs/editor/various/block-deletion.spec.js

```diff
@@ -0,0 +1,350 @@
+			attributes: { content: 'Second' },
+		} );
+
+		// Leave last paragraph empty and ensure it's focused.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await expect(
+			editor.canvas.getByRole( 'document', { name: /Empty block/ } )
+		).toBeFocused();
+
+		// Hit backspace to remove the empty paragraph.
+		await page.keyboard.press( 'Backspace' );
+
+		// Ensure the last block was removed.
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second' );
+
+		// Ensure the caret is in a correct position.
+		await page.keyboard.type( '| ← caret was here' );
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second| ← caret was here' );
+	} );
+
+	test( 'deleting last selected block via backspace', async ( {
+		editor,
+		page,
+	} ) => {
+		// Add a couple of paragraphs.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'First' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Second' },
+		} );
+
+		// Add the Image block and focus it.
+		await editor.insertBlock( {
+			name: 'core/image',
+		} );
+		const imageBlock = editor.canvas.getByRole( 'document', {
+			name: 'Block: Image',
+		} );
+		await expect(
+			imageBlock.getByRole( 'button', { name: 'Upload' } )
+		).toBeFocused();
+		await page.keyboard.press( 'ArrowUp' );
+		await expect( imageBlock ).toBeFocused();
+
+		// Hit backspace and ensure the Image block was removed.
+		await page.keyboard.press( 'Backspace' );
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second' );
+
+		// Ensure the caret is in a correct position.
+		await page.keyboard.type( '| ← caret was here' );
+		await expect(
+			editor.canvas
+				.getByRole( 'document', {
+					name: 'Paragraph block',
+				} )
+				.last()
+		).toContainText( 'Second| ← caret was here' );
+	} );
+
+	test( 'deleting last two selected blocks via backspace', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Add a couple of paragraphs.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'First' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Second' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'Third' },
+		} );
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+
+		// Ensure the empty paragraph is focused.
+		await expect(
+			editor.canvas.getByRole( 'document', {
+				name: /Empty block/i,
+			} )
+		).toBeFocused();
+
+		// Select the last two paragraphs.
+		await pageUtils.pressKeyWithModifier( 'shift', 'ArrowUp' );
+		await expect(
+			editor.canvas.locator( 'p.is-multi-selected' )
```

Comment:
This is a bit unfortunate, do we have any accessible way to determine if a block is selected? Let's ditch the `p` if there's not.

```suggestion
			editor.canvas.locator( '.is-multi-selected' )
```

Reply:
I couldn't really find any other indication of the selection. 😕 The `document` role elements don't support the `selected` attribute, so we can't use it in the `getByRole` locator:

```
locator.click: Error: "selected" attribute is only supported for roles: "columnheader", "gridcell", "option", "row", "rowheader", "tab", "treeitem"
```